lib/srv/desktop/rdp/rdpclient/src/client/global.rs

```diff
@@ -78,14 +80,18 @@ impl ClientHandles {
     }
 
     pub fn insert(&self, cgo_handle: CgoHandle, client_handle: ClientHandle) {
-        self.map.write().insert(cgo_handle, client_handle);
+        self.map.lock().unwrap().insert(cgo_handle, client_handle);
     }
 
     pub fn get(&self, cgo_handle: CgoHandle) -> Option<ClientHandle> {
-        self.map.read().get(&cgo_handle).map(|c| (*c).clone())
+        self.map
+            .lock()
+            .unwrap()
```

Comment:
Should we do `unwrap_or_else(PoisonError::into_inner)` instead? I don't believe that any of the code can currently panic while holding the lock, but things can change.

Reply:
Do you mean something like this?

```
    pub fn get(&self, cgo_handle: CgoHandle) -> Option<ClientHandle> {
        self.map.lock().ok()?.get(&cgo_handle).map(|c| (*c).clone())
    }
```

Reply:
No, that will forever result in `get` not finding things if the mutex actually ends up being poisoned. Using `PoisonError::into_inner` gets you the mutex guard ignoring any poisoning, which would match the existing behavior of the `parking_lot` mutex, and shouldn't really happen anyway.

That being said, if we're confident that it won't happen (and we are) there's also no penalty to using `unwrap`.